src/client/containers/FavoritePage/FavoritePage.Container.js

```diff
@@ -0,0 +1,78 @@
+
+import './FavoritePage.styles.css';
+
+   
+import React, { useState, useEffect, useCallback } from "react";
+import FavouriteList from '../../components/FavouriteList/FavouriteList.component';
+
+
+
+export const addProductToFavorites = (user_id,product_id) => {
+  console.log("in add favorites container");
+  console.log(user_id,product_id)
+ 
+  const requestOptions = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        product_id:product_id,
+        user_id: user_id
+        
+       }),
+       
+    };
+    fetch("/api/favorites/add", requestOptions)
+      .then((response) => response.json())
+      .then((result) => {
+        console.log(result);
+      });
+  };
+
+export default function FavoritePage  () {
+  
+  
+  const [favorites, setFavorites] = useState([]);
+
+
+  const GetFavorites = useCallback(() => {
+    let apiUrl = "/api/users/2/favorites";
+    console.log(apiUrl);
+    fetch(apiUrl)
+      .then((res) => res.json())
+      .then((result) => {
+        console.log(result);
+
```

Comment:
The result you caan console.log if you want to see what you get back from the api .... but we need to use it ot setFavorites(result) . Like `  const GetFavorites = useCallback(() => {
    const apiUrl = `/api/users/2/favorites`;
    fetch(apiUrl)
      .then((res) => res.json())
      .then((result) => {
        setFavorites(result);
      });
  }, []);
  console.log(favorites);

  useEffect(() => {
    GetFavorites();
  }, [GetFavorites]);`

Reply:
It is  not working.....

Reply:
This is only a direction we can go on  because we have to find a better solution then console.log(result) so it needs improving. 👍

Reply:
Hey, the previous one was also working (setFavorites was present but an extra work was being done). But I like the suggestion and have made changes accordingly.